megatron/model/fused_layer_norm.py

```diff
@@ -56,8 +59,12 @@ def __init__(self, normalized_shape, eps=1e-5,
             normalized_shape = (normalized_shape,)
         self.normalized_shape = torch.Size(normalized_shape)
         self.eps = eps
-        self.weight = Parameter(torch.Tensor(*normalized_shape))
-        self.bias = Parameter(torch.Tensor(*normalized_shape))
+        self.weight = Parameter(torch.empty(*normalized_shape,
```

Comment:
Am I correct that this change will create device tensors instead of host tensors in the current code? I am concerned that this could break BC by introducing unexpected device OOMs.

Reply:
eventually while running the model on device weights are moved. why will it use extra device memory or break any BC ?

Reply:
Apologies that I don't remember the flow clearly. My concern is that currently when the weights are moved to device, it is the sharded and/or half'd weights that moved. That will have different device memory footprint compared to the new code that seems to instantiate the unshared and fp32 weights on device. Am I correct?

Reply:
Norm layers weights are never sliced. we apply sequence parallelism on sequence dim of input.

Reply:
Thanks for the correction that norm layers are not TP-sliced. I believe they are half'd along with other parameters, no? 

Can you please share the motivation for moving to device at this point? Thanks!

Reply:
no, norm layer weights are never TP sliced.
HPU device is based on lazy mode execution, tensor attributes are lost(if set on cpu) after moving to device.

Reply:
I already understood that norm weights are not TP-sliced, by half'd I meant 16bit precision, not TP-slicing :). 

Thanks for sharing the rationale for moving to device during initialization. I recall that a similar issue showed up with ZeRO initialization that required a non-trivial solution from your team. 

How about restricting this behavior only to HPU with something like 
```python
   init_device = get_accelerator().current_device_name() if get_accelerator().name == 'hpu' else None
```

Reply:
i will do this for hpu only. regarding 16bit usage - that's why i am using args.params_dtype

Reply:
please review now